src/__tests__/mocks/SpyConnectionBehaviour.ts

```diff
@@ -13,7 +13,9 @@ import MBSpecs from '../../script/microbit-interfacing/MBSpecs';
  */
 class SpyConnectionBehaviour implements ConnectionBehaviour {
   onIdentifiedAsOutdated(): void {
-    throw new Error('Method not implemented.');
+    // This happens in the tests due to async code that's not waited
+    // for. The test Bluetooth code doesn't ever send a version.
+    // Making a no-op for now.
```

Comment:
This was necessary to get the tests passing as Vitest flags background errors. The particular error I saw was from the timeout that waits for a version to be sent which is 4_000 ms. So if the tests took > 4_000 ms then they error. I tried better fixes but they were a deep rabbit hole. I'll try to revisit upstream.